workbench/src/main/setupChangeLanguage.js

```diff
@@ -1,16 +1,25 @@
-import i18n from 'i18next';
-import { ipcMain } from 'electron';
+import Store from 'electron-store';
+import { app, ipcMain } from 'electron';
 import { getLogger } from './logger';
 import { ipcMainChannels } from './ipcMainChannels';
 
 const logger = getLogger(__filename.split('/').slice(-1)[0]);
 
+const store = new Store();
+
 export default function setupChangeLanguage() {
+  ipcMain.on(ipcMainChannels.GET_LANGUAGE, (event) => {
+    // default to en if no language setting exists
+    event.returnValue = store.get('language', 'en');
+  });
+
   ipcMain.handle(
     ipcMainChannels.CHANGE_LANGUAGE,
     (e, languageCode) => {
       logger.debug('changing language to', languageCode);
-      i18n.changeLanguage(languageCode);
+      store.set('language', languageCode);
+      app.relaunch();
+      app.exit();
```

Comment:
Hey @emlys I think you may have missed my big comment about this `app.exit()` vs. `app.quit()`. Sorry I didn't comment in-line, it all occurred to me as I was testing things out after I had reviewed the code. Please see my big comment and let me know what you think about mitigating leftover processes. Thanks!

Reply:
Ah sorry I missed that. I agree that it would be better to `quit()` and have the cleanup run, but in my testing `quit()` does not actually close the window. You end up with duplicate workbench windows. (At least on mac - the latest binary artifacts on this branch have that change, if you'd like to try it). I can look into why that's happening, but it seemed like something that might be outside our control.

Reply:
@emlys Ahh interesting, so maybe it is not so simple as switching `exit()` for `quit()`. Nevertheless, I think we do need to come with an approach to make sure the python processes are cleaned up, or not duplicated on startup if we wish to leave them open.

Reply:
BTW we have a `/ready` endpoint for checking if the flask app is running or not, so that could be handy.

Reply:
@davemfish I think this is a bug in the existing `before-quit` handler. When I quit the app normally, the python processes are not cleaned up. It gets stuck in an infinite loop of `shutdownPythonProcess` calls. Same thing is happening when the app quits programatically.

Reply:
You were right, this was happening because of an old python process. When there's only the expected process, the cleanup succeeds.